lib/safira/interaction/interaction.ex

```diff
@@ -139,4 +142,57 @@ defmodule Safira.Interaction do
         {:error, changeset}
     end
   end
+
+  @doc """
+  Returns the only existing Spotlight
+  """
+  def get_spotlight() do
+    Repo.all(Spotlight) |> List.first()
+  end
+
+  @doc """
+  Starts the spotlight
+  """
+  def start_spotlight(company) do
+    Multi.new()
+    # get or build a spotlight
+    |> Multi.run(:get_spotlight, fn _repo, _changes ->
+      {:ok,
+       get_spotlight() ||
+         %Spotlight{}}
+    end)
+    # update the spotlight
+    |> Multi.insert_or_update(:upsert_spotlight, fn %{get_spotlight: spotlight} ->
+      Spotlight.changeset(spotlight, %{badge_id: company.badge_id, active: true})
+    end)
+    # update company's remaining_spotlights
+    |> Multi.update(
+      :update_company,
+      Company.start_spotlight_changeset(company,
+      %{remaining_spotlights: company.remaining_spotlights - 1})
+    )
+    |> Repo.transaction()
+    |> case do
+      {:ok, result} ->
+        {:ok, result}
+
+      {:error, _failed_operation, changeset, _changes_so_far} ->
+        {:error, changeset}
+    end
```

Comment:
These are not the only possible cases. It is possible that concurrent updates when being updated throw an Ecto.StaleEntryError. So you need to handle this case, otherwise a 500 internal server error will occur.